// @radix-ui/react-popper/dist/index.module.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index_module$6 = require('../../react-arrow/dist/index.module.js');
var index_module$2 = require('../../react-primitive/dist/index.module.js');
var index_module$4 = require('../../react-use-size/dist/index.module.js');
var index_module$3 = require('../../react-use-rect/dist/index.module.js');
var index_module = require('../../react-context/dist/index.module.js');
var index_module$1 = require('../../react-compose-refs/dist/index.module.js');
var index_module$5 = require('../../popper/dist/index.module.js');
var React = require('react');
var _extends = require('../../../@babel/runtime/helpers/esm/extends.js');

function _interopNamespace(e) {
	if (e && e.__esModule) return e;
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n["default"] = e;
	return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);

const [c,l]=index_module.createContextScope("Popper");const[f,d]=c("Popper");const Popper=e=>{const{__scopePopper:o,children:r}=e,[t,n]=React__namespace.useState(null);return React__namespace.createElement(f,{scope:o,anchor:t,onAnchorChange:n},r)};const PopperAnchor=/*#__PURE__*/React__namespace.forwardRef(((e,r)=>{const{__scopePopper:t,virtualRef:n,...p}=e,c=d("PopperAnchor",t),l=React__namespace.useRef(null),f=index_module$1.useComposedRefs(r,l);return React__namespace.useEffect((()=>{c.onAnchorChange((null==n?void 0:n.current)||l.current);})),n?null:/*#__PURE__*/React__namespace.createElement(index_module$2.Primitive.div,_extends["default"]({},p,{ref:f}))}));const[u,m]=c("PopperContent");const PopperContent=/*#__PURE__*/React__namespace.forwardRef(((e,n)=>{const{__scopePopper:c,side:l="bottom",sideOffset:f,align:m="center",alignOffset:w,collisionTolerance:h,avoidCollisions:x=!0,...v}=e,P=d("PopperContent",c),[A,g]=React__namespace.useState(),E=index_module$3.useRect(P.anchor),[y,C]=React__namespace.useState(null),S=index_module$4.useSize(y),[R,O]=React__namespace.useState(null),_=index_module$4.useSize(R),b=index_module$1.useComposedRefs(n,(e=>C(e))),z=function(){const[e,o]=React__namespace.useState(void 0);return React__namespace.useEffect((()=>{let e;function r(){o({width:window.innerWidth,height:window.innerHeight});}function t(){window.clearTimeout(e),e=window.setTimeout(r,100);}return r(),window.addEventListener("resize",t),()=>window.removeEventListener("resize",t)}),[]),e}(),T=z?DOMRect.fromRect({...z,x:0,y:0}):void 0,{popperStyles:k,arrowStyles:L,placedSide:B,placedAlign:D}=index_module$5.getPlacementData({anchorRect:E,popperSize:S,arrowSize:_,arrowOffset:A,side:l,sideOffset:f,align:m,alignOffset:w,shouldAvoidCollisions:x,collisionBoundariesRect:T,collisionTolerance:h}),H=void 0!==B;return React__namespace.createElement("div",{style:k,"data-radix-popper-content-wrapper":""},/*#__PURE__*/React__namespace.createElement(u,{scope:c,arrowStyles:L,onArrowChange:O,onArrowOffsetChange:g},/*#__PURE__*/React__namespace.createElement(index_module$2.Primitive.div,_extends["default"]({"data-side":B,"data-align":D},v,{style:{...v.style,animation:H?void 0:"none"},ref:b}))))}));const PopperArrow=/*#__PURE__*/React__namespace.forwardRef((function(o,r){const{__scopePopper:t,offset:n,...i}=o,p=m("PopperArrow",t),{onArrowOffsetChange:c}=p;return React__namespace.useEffect((()=>c(n)),[c,n]),/*#__PURE__*/React__namespace.createElement("span",{style:{...p.arrowStyles,pointerEvents:"none"}},/*#__PURE__*/React__namespace.createElement("span",{ref:p.onArrowChange,style:{display:"inline-block",verticalAlign:"top",pointerEvents:"auto"}},/*#__PURE__*/React__namespace.createElement(index_module$6.Root,_extends["default"]({},i,{ref:r,style:{...i.style,display:"block"}}))))}));const Root=Popper;const Anchor=PopperAnchor;const Content=PopperContent;const Arrow=PopperArrow;

exports.Anchor = Anchor;
exports.Arrow = Arrow;
exports.Content = Content;
exports.Popper = Popper;
exports.PopperAnchor = PopperAnchor;
exports.PopperArrow = PopperArrow;
exports.PopperContent = PopperContent;
exports.Root = Root;
exports.createPopperScope = l;
